src/entity/posts/ActivePost.ts

```diff
@@ -0,0 +1,131 @@
+import { ViewColumn, DataSource, ViewEntity } from 'typeorm';
+import { Post, PostOrigin, PostType } from './Post';
+import { Toc } from './ArticlePost';
+
+@ViewEntity({
+  expression: (connection: DataSource) =>
+    connection
+      .createQueryBuilder()
+      .select('p.*')
+      .from(Post, 'p')
+      .where('p.deleted = false')
+      .andWhere('p.visible = true')
+      .andWhere(`p."sourceId" != 'unknown'`),
+})
+export class ActivePost {
+  @ViewColumn()
```

Comment:
Do we really need to define all these fields? I think we can get away without it

Reply:
I agree, just not sure right now which ones are needed and had some trouble with some missing columns earlier so I stashed them all in.